clarifai/client/nodepool.py

```diff
@@ -0,0 +1,197 @@
+import os
+from typing import Any, Dict, Generator, List
+
+import yaml
+from clarifai_grpc.grpc.api import resources_pb2, service_pb2
+from clarifai_grpc.grpc.api.status import status_code_pb2
+from google.protobuf.json_format import MessageToDict
+
+from clarifai.client.base import BaseClient
+from clarifai.client.deployment import Deployment
+from clarifai.client.lister import Lister
+from clarifai.errors import UserError
+from clarifai.utils.logging import get_logger
+
+
+class Nodepool(Lister, BaseClient):
+  """Nodepool is a class that provides access to Clarifai API endpoints related to Nodepool information."""
+
+  def __init__(self,
+               nodepool_id: str = None,
+               user_id: str = None,
+               base_url: str = "https://api.clarifai.com",
+               pat: str = None,
+               token: str = None,
+               root_certificates_path: str = None,
+               **kwargs):
+    """Initializes a Nodepool object.
+
+    Args:
+        url (str): The URL to initialize the dataset object.
+        nodepool_id (str): The Nodepool ID for the Nodepool to interact with.
+        base_url (str): Base API url. Default "https://api.clarifai.com"
+        pat (str): A personal access token for authentication. Can be set as env var CLARIFAI_PAT
+        token (str): A session token for authentication. Accepts either a session token or a pat. Can be set as env var CLARIFAI_SESSION_TOKEN
+        root_certificates_path (str): Path to the SSL root certificates file, used to establish secure gRPC connections.
+        **kwargs: Additional keyword arguments to be passed to the Dataset.
```

Comment:
Update this for Nodepool and similarly in the deployment file

Reply:
Done